File: Numerical Methods/Python/gaussian_integration.py
import math

# Define the function f(x)
def f(x):
    return x**3 + 1  # Given function

# Function to perform 2-point Gaussian Quadrature
def gaussian_quadrature(a, b):
    # Gaussian points
    z1, z2 = -0.57735, 0.57735  

    # Transform integration limits
    x1 = ((b - a) / 2) * z1 + (b + a) / 2
    x2 = ((b - a) / 2) * z2 + (b + a) / 2

    # Gaussian Quadrature Formula
    result = ((b - a) / 2) * (f(x1) + f(x2))

    return result

def main():
    # User input
    a = float(input("Enter Lower Limit: "))
    b = float(input("Enter Upper Limit: "))

    result = gaussian_quadrature(a, b)
    
    print(f"\nValue of Integration = {result:.6f}")

if __name__ == "__main__":
    main()

File: Numerical Methods/C-programming/gaussian integration.cpp
#include <stdio.h>
#include <math.h>

#define f(x) ((x) * (x) * (x) + 1) // Ensure correct macro expansion

int main() {
    float a, b, x1, x2, v;
    const float z1 = -0.57735, z2 = 0.57735; // Gaussian points

    // User input
    printf("Enter Lower and Upper Limit: ");
    scanf("%f %f", &a, &b);

    // Transform integration limits
    x1 = ((b - a) / 2) * z1 + (b + a) / 2;
    x2 = ((b - a) / 2) * z2 + (b + a) / 2;

    // Gaussian Quadrature Formula
    v = ((b - a) / 2) * (f(x1) + f(x2));

    // Display result
    printf("\nValue of Integration = %.6f\n", v);
	getchar();
   getchar();
    return 0;
}

